### PR TITLE
Add test for split varint decoding

### PR DESCRIPTION
**Pull request by jaskij**
An additional test to force testing non-contguous chunks in `Buf`. This is necessary, since this edge case is, theoretically, supported by the `Buf` trait.

Additionally, this also serves to test the slow path decoding.

Although, as https://github.com/tokio-rs/bytes/issues/474 shows, this is probably a low-priority case.

### CODE REVIEW

File: src/encoding.rs [hunk 1]
```diff
@@ -1776,4 +1776,33 @@ mod test {
         (String, string),
         (Vec<u8>, bytes)
     ]);
+
+
+    #[test]
+    fn split_varint_decoding() {
```
Comment (caspermeijn) [inline src/encoding.rs:1782]:
```suggestion
    /// `decode_varint` accepts a `Buf`, which can be multiple concatinated buffers. 
    /// This test ensures that future optimizations don't break the 
    /// `decode_varint` for non-continuous memory.
    fn split_varint_decoding() {
```

**Review by caspermeijn — CHANGES_REQUESTED** [review]
Looks good to me. Can you rebase to latest master?
